src/android/webview.kt

```diff
@@ -84,6 +91,50 @@ open class WebViewClient (activity: WebViewActivity) : android.webkit.WebViewCli
   ): android.webkit.WebResourceResponse? {
     var url = request.url
 
+    // should be set in window loader
+    assert(rootDirectory.length > 0)
+
+    // look for updated resources in ${pwd}/files
+    // live update systems can write to /files (/assets is read only)
+    if (url.host == "appassets.androidplatform.net" && url.pathSegments.get(0) == "assets") {
+      var first = true
+      val filePath = StringBuilder(rootDirectory)
+      for (item in url.pathSegments) {
+        if (!first) {
+          filePath.append("/${item}")
+        }
+        first = false
+      }
+
+      val file = java.io.File(filePath.toString())
+      if (file.exists() && !file.isDirectory()) {
+        var input = java.io.FileInputStream(file)
+        var bytes = input.readAllBytes()
+        input.close()
+        val stream = java.io.PipedOutputStream()
+        kotlin.concurrent.thread {
+          stream.write(bytes, 0, bytes.size)
+          stream.close()
+        }
+
+        val response = android.webkit.WebResourceResponse(
+          if (filePath.toString().endsWith(".js")) "text/javascript" else "text/html",
+          "utf-8",
+          java.io.PipedInputStream(stream)
```

Comment:
Why not just use `input` here instead of buffer the entire stream above to use here

Reply:
Agree, old code from when the buffer was being modified.